### PR TITLE
Refactor hamburgerbutton functionality

### PR DESCRIPTION
**Pull request by NathanCorbin**
**Description**
Refactor hamburger button  functionality and added menu tests

**Motivation and Context**
Old functionality was ugly

**Checklist (check all that is appropriate)**
- [ ] Issue is linked in this PR
- [x] Existing tests pass
- [ ] Accessibility tested (if appropriate)

### CODE REVIEW

File: src/components/navigation/menu/HamburgerButton.module.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
+@import '../../../styles/shared/variables';
+
+.hamburgerIcon {
+  color: '#70F3FF';
```
Comment (QuentinGuenther) [inline src/components/navigation/menu/HamburgerButton.module.scss:4]: why not re use the color variables?

File: src/components/navigation/menu/HamburgerButton.module.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  display: block;
+  position: absolute;
+  right: 0;
+  top: 0;
+  padding: 0;
+}
+
+.hamburgerButton {
```
Comment (QuentinGuenther) [inline src/components/navigation/menu/HamburgerButton.module.scss:12]: is this to reset button styles? If so these should be a `button-reset` mixin and `@included` here

File: src/components/navigation/menu/__tests__/Menu.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { shallow } from 'enzyme'
+import toJson from 'enzyme-to-json'
+import Menu from '../Menu'
+
+describe('Menu', () => {
```
Comment (QuentinGuenther) [inline src/components/navigation/menu/__tests__/Menu.spec.tsx:6]:
```suggestion
describe('<Menu />', () => {
```

File: src/components/navigation/menu/__tests__/HamburgerButton.spec.tsx [hunk 4]
```diff
@@ -6,13 +6,13 @@ import HamburgerButton from '../HamburgerButton'
 describe('HamburgerButton', () => {
```
Comment (QuentinGuenther) [inline src/components/navigation/menu/__tests__/HamburgerButton.spec.tsx:6]:
```suggestion
describe('<HamburgerButton />', () => {
```

File: src/components/navigation/menu/HamburgerButton.tsx [hunk 5]
```diff
@@ -1,38 +1,16 @@
...
-}
+import HamburgerStyles from './HamburgerButton.module.scss'
 
 const HamburgerButton: React.FC = () => {
-  const [isMenuDisplayed, setIsMenuDisplayed] = React.useState(false)
-  const changeMenuDisplay = () => {
-    setIsMenuDisplayed(!isMenuDisplayed)
-  }
-
+  const [open, isOpen] = React.useState(false)
```
Comment (QuentinGuenther) [inline src/components/navigation/menu/HamburgerButton.tsx:7]: try this for convention
```
const [isOpen, setIsOpen] = React.useState(false)
```

File: src/components/navigation/menu/HamburgerButton.module.scss (hunk 2 above)
Comment (NathanCorbin) [reply src/components/navigation/menu/HamburgerButton.module.scss:12]: It's the styles for the actual button element that wraps around img tag

**Pull request merged**